org.civicrm.hremergency/tests/phpunit/api/v3/Contact/DeleteemergencycontactTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use Tests\CiviHR\HREmergency\Fabricator\EmergencyContactFabricator;
+
+/**
+ * @group headless
+ */
+class api_v3_Contact_DeleteEmergencyContactTest extends \PHPUnit_Framework_TestCase
+  implements HeadlessInterface, TransactionalInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  public function testDeletingNonExistentContact() {
+    $message = "Could not find emergency contact with ID '0'";
+    $this->setExpectedException(CiviCRM_API3_Exception::class, $message);
+    civicrm_api3('Contact', 'deleteemergencycontact', ['id' => 0]);
+  }
+
+  public function testDeletingWithoutPermission() {
+    $contacts = civicrm_api3('Contact', 'get');
+    $contactIDs = array_column($contacts['values'], 'id');
+    $sampleID = current($contactIDs);
+
+    $id = EmergencyContactFabricator::fabricate($sampleID, 'Kevin Foot');
+    $message = "Only an emergency contacts' relation can delete them";
+    $this->setExpectedException(CiviCRM_API3_Exception::class, $message);
+    civicrm_api3('Contact', 'deleteemergencycontact', ['id' => $id]);
+  }
+
+  public function testDeletionOfOwnContact() {
+    $contacts = civicrm_api3('Contact', 'get');
+    $contactIDs = array_column($contacts['values'], 'id');
+    $sampleID = current($contactIDs);
+
+    $this->registerCurrentLoggedInContactInSession($sampleID);
+    $id = EmergencyContactFabricator::fabricate($sampleID, 'Kevin Foot');
+
+    $result = civicrm_api3('Contact', 'deleteemergencycontact', ['id' => $id]);
+
+    $this->assertEquals(0, $result['is_error']);
+  }
+
+  /**
+   * @param $contactID
+   */
+  protected function registerCurrentLoggedInContactInSession($contactID) {
+    $session = CRM_Core_Session::singleton();
+    $session->set('userID', $contactID);
+  }
```

Comment:
Please use the helper from HRCore instead: https://github.com/civicrm/civihr/blob/staging/uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Test/Helpers/SessionHelpersTrait.php

Reply:
I was aware of this but was previously told in reviews to copy code over from other extensions if it didn't exist in the extension being tested.

Reply:
mmmm... I wonder if it was me who said that? I don't remember, but perhaps, depending on the context, it made more sense at the time? Perhaps it was because this was implemented in L&A, which wasn't merged into staging yet? Anyway, I agree with you and this duplication doesn't look good. I'll try to consolidate them as part of the Jenkins work.

As for using fabricators from other extensions, please install the extension using the `setupHeadless()` method. The idea is to have a single extension, but when don't know if that will really happen nor when. The same way, there's no guarantee that, perhaps, we decide to move or rename the hrcore extension.